pkg/datasource/data.go

```diff
@@ -137,12 +138,44 @@ func (ds *dataSource) NewData() Data {
 	d := &data{
 		Payload: make([][]byte, ds.payloadCount),
 	}
-	for i := range d.Payload {
-		d.Payload[i] = make([]byte, 0)
+
+	// Allocate memory for fixed size fields added with Add{Sub}Field
```

Comment:
I prefer to be as explicit (and over cautious) as possible here.

Reply:
Fields added with that function don't need memory to be preallocated as the setter will set the entire buffer in https://github.com/inspektor-gadget/inspektor-gadget/blob/78a0153a6786d839bbf02cc047fd64ed4c7e5f9b/pkg/datasource/accessors.go#L190 to avoid copying the memory. 

I suppose calling `Set` in a member of a "static field container" without first calling `Set()` on the parent is a bad behavior and something we don't need to protect against. 

All those members have the FieldFlagStaticMember flag set 

https://github.com/inspektor-gadget/inspektor-gadget/blob/309f14ea56a07ccacbde833f9eb988bd23333501/pkg/datasource/data.go#L190

and the parent as FieldFlagContainer

https://github.com/inspektor-gadget/inspektor-gadget/blob/309f14ea56a07ccacbde833f9eb988bd23333501/pkg/datasource/data.go#L253